core/src/main/java/com/novoda/noplayer/internal/mediaplayer/AndroidMediaPlayerImpl.java

```diff
@@ -121,8 +121,11 @@ public void play() throws IllegalStateException {
         requestSurface(new SurfaceHolderRequester.Callback() {
             @Override
             public void onSurfaceHolderReady(SurfaceHolder surfaceHolder) {
+                boolean videoWasNotPlaying = !isPlaying();
```

Comment:
no, we can't. The state gets changed in the next line.

We are storing the `isPlaying` before starting the player internally, which will change the state of `isPlaying` and will result in emitting the `onVideoPlaying`.